src/ngraph/frontend/onnx_import/core/null_node.hpp

```diff
@@ -0,0 +1,48 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>
+
+#include "ngraph/node.hpp"
+#include "ngraph/node_vector.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        /// \brief Represents a missing optional input or output of an ONNX node
+        ///
+        /// Some ONNX operators have inputs or outputs that are marked as optional,
+        /// which means that a referring node MAY forgo providing values for such inputs
+        /// or computing these outputs.
+        /// An empty string is used in place of a name of such input or output.
+        ///
+        /// More:
+        /// https://github.com/onnx/onnx/blob/master/docs/IR.md#optional-inputs-and-outputs
+        class NullNode : public ngraph::Node
+        {
+        public:
+            NullNode();
+
+            virtual bool is_null() const { return true; }
```

Comment:
Please add `override` keyword, see [here why](https://stackoverflow.com/questions/13880205/is-the-override-keyword-just-a-check-for-a-overridden-virtual-method). Moreover the `final` keyword also is suitable. If you let anything derive from `NullNode` then you probably don't want it to change it's behavior of `is_null`, since it no longer will be a `NullNode`.